File: apps/api/internal/models/project.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Project struct {
	ID          uuid.UUID  `json:"id" db:"id"`
	Name        string     `json:"name" db:"name"`
	Description *string    `json:"description,omitempty" db:"description"`
	ThumbnailURL *string   `json:"thumbnail_url,omitempty" db:"thumbnail_url"`
	OwnerID     uuid.UUID  `json:"owner_id" db:"owner_id"`
	CreatedAt   time.Time  `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at" db:"updated_at"`
}

type ProjectMember struct {
	ProjectID uuid.UUID `json:"project_id" db:"project_id"`
	UserID    uuid.UUID `json:"user_id" db:"user_id"`
	Role      string    `json:"role" db:"role"` // owner, editor, viewer
	CreatedAt time.Time `json:"created_at" db:"created_at"`
}

File: apps/web/types/project.ts
export interface Project {
  id: string
  name: string
  description?: string
  thumbnail_url?: string
  owner_id: string
  created_at: string
  updated_at: string
}

export interface Page {
  id: string
  project_id: string
  name: string
  order: number
  created_at: string
  updated_at: string
}

export interface User {
  id: string
  email: string
  name: string
  avatar_url?: string
}
